conda/core/index.py

```diff
@@ -100,6 +102,54 @@ def supplement_index_with_cache(index, channels):
         index[dist] = rec
 
 
+def supplement_index_with_repodata(index, repodata, channel, priority):
```

Comment:
This function replaces the inner loop of `fetch_index.make_index`. I needed to break it out so that it could easily be called from `test_resolve`.

For packages that do not have `with_features_depends`, this produces identical behavior. But for packages that have this entry, it creates extra copies of the same package, but with a modified `Dist` and modified `depends` and `features` fields. This way, `Resolve` just consider it to be a normal package.